Documentation/Functions/Typolink.rst

```diff
@@ -543,27 +543,51 @@ userFunc
     :name: typolink-userFunc
     :type: :ref:`data-type-function-name`
 
-    This passes the link-data compiled by the typolink function to a user-
-    defined function for final manipulation.
-
-    The :php:`$content` variable passed to the user-function (first parameter) is
-    an array with the keys "TYPE", "TAG", "url", "targetParams" and
-    "aTagParams".
-
-    TYPE is an indication of link-kind: mailto, url, file, page
-
-    TAG is the full <A>-tag as generated and ready from the typolink
-    function.
-
-    The actual tag value is constructed like this:
-
-    ..  code-block:: php
-
-        $contents = '<a href="' . $finalTagParts['url'] . '"'
-                   . $finalTagParts['targetParams']
-                   . $finalTagParts['aTagParams'] . '>';
-
-    The userfunction must return an <A>-tag.
+    All of the :typoscript:`typolink` TypoScript configuration will be parsed and evaluated
+    by the TYPO3 Core's :php:`LinkFactory->create()` method, and then passed on
+    to the defined :typoscript:`userFunc` for further manipulation. The :typoscript:`userFunc` needs to
+    return an object implementing the :php:`LinkResultInterface`. The currently calculated
+    typolink is passed as an argument to the :php:`userFunc` as an object of the same type. This allows
+    to return either an enriched link, or a completely new one.
+
+    The detailed execution steps are:
+
+    First, the :typoscript:`typolink` will be created as configured by the specified TypoScript.
+    This will result in an object of Type :php:`LinkResultInterface`. This object receives
+    all of the TypoScript :typoscript:`typolink` configuration as properties, and makes them
+    available via corresponding getters and setters.
```

Comment:
There is at least no "setter" in `LinkResult`. This object is immutable, so it can not be modified. The object contains various `with*` methods to "clone" the existing LinkResult into a new LinkResult with one changed property.

Reply:
Very true. This was a left over from my rewording, thanks for catching this mistake!